### ISSUE TITLE
Error when webbpsf is not installed

### ISSUE THREAD

**Issue by nespinoza** [spacetelescope/awesimsoss#60]
Hi Joe!

 Related to [my previous issue](https://github.com/spacetelescope/awesimsoss/issues/59), turns out that If one doesn't have `webbpsf`, the code tries to do something, but it fails with the following error:

```
~/.local/lib/python3.7/site-packages/awesimsoss-0.3.1-py3.7.egg/awesimsoss/make_trace.py in calculate_psf_tilts()
     51         # Get the y-coordinate of the trace polynomial in this column
     52         # (center of the trace)
---> 53         coeffs = trace_polynomials(subarray=subarray, order=order)
     54         trace = np.polyval(coeffs, X)
     55 

NameError: name 'trace_polynomials' is not defined
```

So effectively the code doesn't work if one doesn't have `webbpsf`. I pressume an `import hotsoss` is needed here with a subsequent `locate_trace.trace_polynomial()` instead of `trace_polynomials`? I haven't looked into `hotsoss` in detail, but I'm sure you know what is going on :-).

 Thanks!
Néstor

PD: Indeed, I tested that adding `from hotsoss import locate_trace` and replacing the line mentioned above solves this issue. Can't push it though, because my username is not listed as part of STScI's Github :-(.

**Comment by hover2pi** [comment]
Thanks for the debugging! I'll add the hotsoss imported function to that line of the code in the next PR!

**Comment by exowanderer** [comment]
Hello @nespinoza & @hover2pi,

I do not see a `trace_polynomials` function in the `locate_trace.py` file, or anywhere on the [awesimsoss](https://github.com/spacetelescope/awesimsoss/search?q=trace_polynomials&unscoped_q=trace_polynomials) + [hotsoss](https://github.com/hover2pi/hotsoss/search?q=trace_polynomials&unscoped_q=trace_polynomials) repos.
Is there a dev version that I am missing?

I tried to add `from hotsoss.locate_trace import trace_polynomials` into the `awesimsoss.make_trace.py` file; but this generated a similar `trace_polynomials not found` error.

Moreover, I have `WebbPSF` installed, which implies to me that this issue is not related to the requirements; unless I misunderstood the first post here.

I am working on a [Colab tutorial based on your README.md file](https://colab.research.google.com/drive/1Qg5FqlfbNrWhh8_C35sjD4gIJCQI1KF7?usp=sharing). I will do the same for the notebooks. I find these are very helpful for people wanting to explore new (to them) packages; and I wanted to contribute to AWESimSOSS again.

Best,

Jonathan

**Comment by nespinoza** [comment]
Hey @exowanderer!

 I can't recall why I said this was something that appeared when you don't have `webbpsf`. Perhaps this was fixed in the latest additions of `awesimsoss`, and now this is a generic problem. 

 As for your `trace_polynomials` question, the `locate_trace` module has a `trace_polynomial` (and not `trace_polynomials`) function in it ([without the "s"](https://github.com/hover2pi/hotsoss/blob/15fda80e7588730bceb8b2c8a8d7d1e6ade0e070/hotsoss/locate_trace.py#L339)). I think I fixed this in my latest PR (https://github.com/spacetelescope/awesimsoss/pull/77) which in commit https://github.com/spacetelescope/awesimsoss/pull/77/commits/2faf39d4397dd25b6cadfcc52241ced6357ee6c1 fixes this issue.

 Let me know if that works*! 

N.

*:  FWIW, [I have a working branch in my own fork](https://github.com/nespinoza/awesimsoss/tree/long-integrations) that incorporates both that PR and the latest PR from Joe (https://github.com/spacetelescope/awesimsoss/pull/73). For some reason, I see a bug when passing the ExoCTK `ModelGrid` to generate limb-darkening (have to write an issue about this; had not had the time yet), but in that branch [I also committed the possibility that the user passes a `list` with limb-darkenings calculated as a function of wavelength](https://github.com/nespinoza/awesimsoss/commit/89daa8a8b4ea88f94a9664c4d9085ed2f9f85662), and then `awesimsoss` takes care to map that to the pixels in the frame.

**Comment by exowanderer** [comment]
> FWIW, [I have a working branch in my own fork](https://github.com/nespinoza/awesimsoss/tree/long-integrations) that incorporates both that PR and the latest PR from Joe (#73). 

That makes me happy that @hover2pi has an active collaborator that he can trust to fork, branch, and PR correctly: he removed my write permissions to our original repo (github.com/hover2pi/awesimsoss) :D :D

**Comment by nespinoza** [comment]
Well, I wouldn't trust myself to write anything either here or there without some review --- that's why I did my own fork, so I can mess up things over there in random branches :).

**Comment by hover2pi** [comment]
I think this is fixed in #73 but let me know if the issue persists @exowanderer and @nespinoza ! Sorry about the permissions Jonathan! Just trying to follow best practices. ITSD says you need to fork the @spacetelescope version and issue a PR for changes!
